### PR TITLE
feat: Added `--logger` option to the commands.

### PR DESCRIPTION
**Pull request by fboucquez**
fix: logger code improvements. No more console.out, loggers are provided and not statically created giving the user, devs, and apps more options
fix: unit tests logs are hidden by default.

This is a PR from a list of improvements I need to do in order to support the new symbol-network app better. 

To be merged after 1.1.1 release.

### CODE REVIEW

File: presets/mainnet/network.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ minVotingKeyLifetime: 112
 maxVotingKeyLifetime: 360
 votingKeyDesiredLifetime: 360
 votingKeyDesiredFutureLifetime: 60
-lastKnownNetworkEpoch: 467
+lastKnownNetworkEpoch: 488
```
Comment (Jaguar0625) [inline presets/mainnet/network.yml:34]: these configuration changes seem unrelated to adding a logger and shouldn't really be mixed into logging changes imo

File: src/logger/LoggerFactory.ts [hunk 2]
```diff
@@ -14,50 +14,80 @@
...
+                winston.format.printf((i) => `${i.timestamp} ${i.level} ${i.message}`),
+            ),
             options: { flags: 'w' },
-            filename: resolve(LoggerFactory.getLogFileName(id)),
+            filename: resolve(fileName),
             level: 'info',
         });
 
-    public static getLogger(id = LogType.System): Logger {
+    public static getLogger(id: LogType, workingDir = BootstrapUtils.defaultWorkingDir): Logger {
```
Comment (Jaguar0625) [inline src/logger/LoggerFactory.ts:56]: wouldn't it be more natural to allow user to specify transports like --log Console|File instead of creating bespoke combinations?

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]

File: presets/mainnet/network.yml (hunk 1 above)
Comment (fboucquez) [reply presets/mainnet/network.yml]: removed, those numbers are automatically updated when running the tests.

File: src/logger/LoggerFactory.ts (hunk 2 above)
Comment (fboucquez) [reply src/logger/LoggerFactory.ts]: Linux confuses '|' with the pipe operation. I've used ',' instead. 
Alternately, we could use --logger console --logger file

**Comment by fboucquez** [comment]
@Jaguar0625 , feedback fixes pushed. Ready to review. More PRs comming

File: src/logger/LoggerFactory.ts [hunk 3]
```diff
@@ -14,50 +14,82 @@
...
     }
 
-    private static getLogFileName(id: LogType): string {
-        switch (id) {
-            case LogType.Audit:
-                return 'auditLogs.log';
-            case LogType.System:
-            default:
-                return 'logs.log';
+    public static getLoggerFromTypes(logTypes: LogType[], workingDir = BootstrapUtils.defaultWorkingDir): Logger {
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply src/logger/LoggerFactory.ts:71]: I was considering apps that could use bootstrap as lib using the type-safe `getLoggerFromTypes`

File: src/logger/LogType.ts [hunk 4]
```diff
@@ -15,6 +15,15 @@
  */
 
 export enum LogType {
-    Audit = 'AUDIT_LOG',
-    System = 'SYSTEM_LOG',
+    // Simple console log without line info
+    Console = 'Console',
+    // Console log with line info
+    ConsoleLog = 'ConsoleLog',
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply src/logger/LogType.ts:21]: `Console` is a replacement for the barebone `console.log`. It is used in commands that may have "prompts" like `wizard`, `encrypt`, `decrypt`, etc.  I'm not 100% sure either if we should have both options or just `ConsoleLog` that adds line information.

Examples:
image
image

What do you think? Should we unify both into `ConsoleLog` (and replace the name to `Console`)?

File: test/logger/LoggerFactory.test.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { expect } from 'chai';
+import { it } from 'mocha';
+import { LoggerFactory } from '../../src';
+
+describe('LoggerFactory', () => {
+    it('BootstrapUtils getLogger file', async () => {
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply test/logger/LoggerFactory.test.ts:6]: Nop :), removed

File: test/logger/LoggerFactory.test.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        expect(transports[0].name).equals('console');
+    });
+
+    it('BootstrapUtils getLogger console log', async () => {
+        const logger = LoggerFactory.getLogger('consoleLOG');
+        const transports = (logger as any)['transports'];
+        expect(transports.length).equals(1);
+        expect(transports[0].name).equals('console');
+    });
+    it('BootstrapUtils getLogger silent', async () => {
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply test/logger/LoggerFactory.test.ts:27]: fixed

File: CHANGELOG.md [hunk 7]
```diff
@@ -12,6 +12,8 @@ The changelog format is based on [Keep a Changelog](https://keepachangelog.com/e
 | ---------------- | ------- | ------------------------------------------------------------------ |
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply CHANGELOG.md:12]: Fixed

File: test/commands/decrypt.test.ts [hunk 8]
```diff
@@ -15,70 +15,65 @@
...
 describe('decrypt', () => {
-    test.add('remove target', () => BootstrapUtils.deleteFolder('target/tests.encrypt'))
+    test.add('remove target', () => BootstrapUtils.deleteFolder(logger, 'target/tests.encrypt'))
         .stdout()
         .command('decrypt --source test/encrypt/encrypted.yml --destination target/tests.encrypt/plain.yml --password 1111'.split(' '))
         .it('runs decrypt and creates file', async (ctx) => {
             expect(ctx.stdout).to.contain(
                 'Decrypted file target/tests.encrypt/plain.yml has been created! Any private keys on this file are now in plain text. Remember to remove the file!',
             );
-            expect(existsSync('target/tests.encrypt/plain.yml')).eq(true);
```
[the remark this answers is not in the document]
Comment (fboucquez) [reply test/commands/decrypt.test.ts]: No idea, added it back.

File: src/logger/LogType.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (fboucquez) [reply src/logger/LogType.ts:21]: I'm tenting to unify the Console options...

[the remark this answers is not in the document]
Comment (Jaguar0625) [reply src/logger/LogType.ts:21]: yea, i think unifying them is fine

File: src/logger/LoggerFactory.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply src/logger/LoggerFactory.ts:71]: ok

**Review by Jaguar0625 — DISMISSED** [review]
